### PR TITLE
Gruuls lair Scripts

### PR DESCRIPTION
**Pull request by aaron02**
**Description**
Scripts for Gruuls lair

**Todo / Checklist**
- [x] Target is branch develop.
- [x] Detailed description about this pull request.
- [x] Checked AE-Coding standards.

**Tests Performed:** 
- [x] Build AE.
- [x] Server startup.
- [x] Log into world.

### CODE REVIEW

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 1]
```diff
@@ -4,34 +4,221 @@ This file is released under the MIT license. See README-MIT for more information
...
+            case NPC_GRUUL_THE_DRAGONKILLER:
+                mGruul = pCreature;
+                break;
+            default:
+                break;
+        }
+
+        if (mMaulgar && mMaulgar->GetScript())
+        {
+            if (mKrosh  && mKrosh->GetScript() && !mKrosh->GetScript()->getLinkedCreatureAIScript())
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:79]: Double space

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 2]
```diff
@@ -4,34 +4,221 @@ This file is released under the MIT license. See README-MIT for more information
...
+                mKrosh->GetScript()->setLinkedCreatureAIScript(mMaulgar->GetScript());
+
+            if (mOlm  && mOlm->GetScript() && !mOlm->GetScript()->getLinkedCreatureAIScript())
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:82]: Double space

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 3]
```diff
@@ -4,34 +4,221 @@ This file is released under the MIT license. See README-MIT for more information
...
+                mOlm->GetScript()->setLinkedCreatureAIScript(mMaulgar->GetScript());
+
+            if (mKiggler  && mKiggler->GetScript() && !mKiggler->GetScript()->getLinkedCreatureAIScript())
+                mKiggler->GetScript()->setLinkedCreatureAIScript(mMaulgar->GetScript());
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:86]: Double space

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 4]
```diff
@@ -4,34 +4,221 @@ This file is released under the MIT license. See README-MIT for more information
...
+
+            if (mBlindeye  && mBlindeye->GetScript() && !mBlindeye->GetScript()->getLinkedCreatureAIScript())
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:88]: Double space

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 5]
```diff
@@ -4,34 +4,221 @@ This file is released under the MIT license. See README-MIT for more information
...
+                mBlindeye->GetScript()->setLinkedCreatureAIScript(mMaulgar->GetScript());
+        }
+    }
+
+    void OnEncounterStateChange(uint32_t entry, uint32_t state) override
+    {
+        switch (entry)
+        {
+            case NPC_MAULGAR:
+                switch (state)
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:98]: Cases with more than one line deserve { } too

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 6]
```diff
@@ -4,34 +4,221 @@ This file is released under the MIT license. See README-MIT for more information
...
+                        break;
+                    case Finished:
+                        setLocalData(DATA_DOOR_MAULGAR, ACTION_ENABLE);
+                        break;
+                    default:
+                        break;
+                }
+                break;
+            case NPC_GRUUL_THE_DRAGONKILLER:
+                switch (state)
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:114]: Cases with more than one line deserve { } too

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 7]
```diff
@@ -4,34 +4,221 @@ This file is released under the MIT license. See README-MIT for more information
...
+                break;
+        }
+    }
+
+    void setLocalData(uint32_t type, uint32_t data) override
+    {
+        switch (type)
+        {
+            case DATA_DOOR_MAULGAR:
+                if (mDoorMaulgar)
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:172]: Cases with more than one line deserve { } too

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 8]
```diff
@@ -4,34 +4,221 @@ This file is released under the MIT license. See README-MIT for more information
...
+                {
+                    if (data == ACTION_ENABLE)
+                        mDoorMaulgar->setState(GO_STATE_OPEN);
+                    else
+                        mDoorMaulgar->setState(GO_STATE_CLOSED);
+                }
+                break;
+            case DATA_DOOR_GRUUL:
+                if (mDoorGruul)
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:181]: Cases with more than one line deserve { } too

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 9]
```diff
@@ -4,34 +4,221 @@ This file is released under the MIT license. See README-MIT for more information
...
+                        mDoorGruul->setState(GO_STATE_CLOSED);
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+public:
+    GruulsLairInstanceScript* Instance;
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:195]: Initialisation?

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp [hunk 10]
```diff
@@ -45,528 +232,433 @@ class LairBruteAI : public CreatureAIScript
...
+        if (actionId == ACTION_ADD_DEATH)
+            sendRandomDBChatMessage(emoteVector, nullptr);
     }
 
-    uint32_t mAliveAdds;
-    int32_t mLastYell;
-    CreatureAISpells* mEnrage;
+protected:
+    std::vector<uint32_t> emoteVector;
+    CreatureAISpells* mArcingSmash;
```
Comment (Zyres) [inline src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:329]: Initialisation?

**Review by Zyres — CHANGES_REQUESTED** [review]

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp (hunk 9 above)
Comment (aaron02) [reply src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:195]: Line 15?

File: src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp (hunk 10 above)
Comment (aaron02) [reply src/scripts/InstanceScripts/Tbc/Raid_GruulsLair.cpp:329]: line 291 emoteVector.clear()
or if u meant mArcingSmash then line 273

**Pull request merged**